src/main/java/edu/hm/hafner/analysis/ModuleDetector.java

```diff
@@ -181,6 +202,56 @@ private String parseBuildXml(final String buildXml) {
         return StringUtils.EMPTY;
     }
 
+    /**
+     * Returns the project name estimated from the build.gradle file path.
+     *
+     * @param buildScript
+     *         Gradle build.gradle file path
+     *
+     * @return the project name or an empty string if the name could not be resolved
+     */
+    private String parseGradle(final String buildScript) {
+        Path path = Paths.get(buildScript).normalize();
+        if (path.getNameCount() >= 2) {
+            Path parent = path.getParent();
+            return parent.getFileName().toString();
+        }
+        return StringUtils.EMPTY;
+    }
+
+    /**
+     * Returns the root project name from the settings.gradle file.
+     *
+     * @param settingsFile
+     *         Gradle settings.gradle file path
+     *
+     * @return the root project override, or an empty string if the name could not be resolved
+     */
+    private String parseGradleSettings(final String settingsFile) {
+        String name = null;
+
+        try (InputStream input = factory.open(settingsFile);
+             Scanner scan = new Scanner(input, "UTF-8")
+        ) {
```

Comment:
Indentation: https://ci.jenkins.io/job/Plugins/job/analysis-model/job/PR-362/6/checkstyle/new/source.a1032ba9-57ef-42c2-9ced-af4ef38c1a44/#234

Reply:
(There are some PMD violations as well: https://ci.jenkins.io/job/Plugins/job/analysis-model/job/PR-362/6/checkstyle/new/source.a1032ba9-57ef-42c2-9ced-af4ef38c1a44/#234)

Reply:
The indentation is weird - because I formatted this using the project rules (or so I thought). It seems the current IntelliJ rules will indent to align to the previous argument, if it started on the same line as the open `(`, but if I put a newline after the `(`, then both lines are indented to 16.

This might be fixable by committing more explicit code style settings to the .idea files?  For now I'll just fix the indentation manually.

Reply:
Also 3/4 of the PMD warnings are addressed. The 4th appears to be an existing warning (unused `NO_RESULT`).

The NPath/Cyclomatic complexity warnings have been addressed by extracting the file-to-module mappings into separate methods, rather than having a bunch of for/if blocks in one method.